fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/Auth.java

```diff
@@ -171,8 +172,8 @@ public void mergeRolesNoCheckName(List<String> roles, Role savedRole) throws Ddl
      */
     public void checkPassword(String remoteUser, String remoteHost, byte[] remotePasswd, byte[] randomString,
             List<UserIdentity> currentUser) throws AuthenticationException {
-        if ((remoteUser.equals(ROOT_USER) || remoteUser.equals(ADMIN_USER)) && remoteHost.equals("127.0.0.1")) {
-            // root and admin user is allowed to login from 127.0.0.1, in case user forget password.
+        if ((remoteUser.equals(ROOT_USER) || remoteUser.equals(ADMIN_USER)) && Config.skip_auth_check) {
```

Comment:
I haven't looked at the calling process, but generally speaking, we recommend constant.equals to prevent NPE.
it's difficult for us to check whether the caller has done the corresponding null value detection.

```
eg: ROOT_USER.equals(remoteUser)
```

Reply:
OK, I will modify this.